libs/core/langchain_core/tools/convert.py

```diff
@@ -421,3 +429,15 @@ def invoke_wrapper(callbacks: Optional[Callbacks] = None, **kwargs: Any) -> Any:
             description=description,
             args_schema=args_schema,
         )
+
+
+class MethodTool:
```

Comment:
can we make this a function instead of a class?

Reply:
I see - this maybe solves the issue of typing. going to suggest some ideas - not sure if they work

Reply:
The class is needed for classmethod support. I have created a wrapper function `methodtool` that simply returns `MethodTool(func)`. How's that?

Reply:
why is it needed for classmethod support?